fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/storage/StorageView.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.storage;
+
+import net.fabricmc.fabric.api.transfer.v1.transaction.Transaction;
+
+/**
+ * A view of a single stored resource in a {@link Storage}, for use with {@link Storage#iterator},
+ * {@link Storage#anyView} or {@link Storage#exactView}.
+ *
+ * <p>A view is always tied to a specific transaction, and should not be accessed outside of it.
+ *
+ * @param <T> The type of the stored resource.
+ */
+public interface StorageView<T> {
+	/**
+	 * Try to extract a resource from this view.
+	 * @return The amount that was extracted.
+	 */
+	long extract(T resource, long maxAmount, Transaction transaction);
+
+	/**
+	 * @return {@code true} if this storage view is empty, or {@code false} otherwise.
+	 */
+	boolean isEmpty();
```

Comment:
```suggestion
default boolean isEmpty() {
   return this.amount() == 0;
}
```

Reply:
I should specify that it's for the resource itself - non-empty resource with amount 0 can have a different semantic than truly empty. (e.g. locked slot)

Reply:
Updated documentation to clarify that it refers to the stored resource.